### PR TITLE
ps: clarify grep example

### PR DESCRIPTION
**Pull request by marcelruland**
Added a note that `ps aux | grep {{string}}` will match the grep process itself unless using a regex with grep. This is something that actively confused me as a newbie.

---

- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message-and-pr-title).
- **Version of the command being documented (if known):**
`ps from procps-ng 4.0.3`
`grep (GNU grep) 3.11`

### CODE REVIEW

File: pages/common/ps.md
Comment (fazlearefin) [inline pages/common/ps.md:16]:
```suggestion
- Search for a process that matches a string (enclose the first letter of searched string in `[]` to filter out `grep` itself):

`ps aux | grep {{'[p]ython3'}}`
```

There is actually this clever trick to enclose the first letter in `[]` to filter out `grep` itself. Let's see what other maintainers think about this change.

Here is a demo:

```console
$ ps aux | grep '[s]leep'
user    3610327  0.0  0.0  18180  1920 pts/1    SN+  20:16   0:00 sleep 1000
$
```

Comment (tricantivu) [reply pages/common/ps.md:16]: Or just use `pgrep`?

Comment (fazlearefin) [reply pages/common/ps.md:16]: `pgrep` does not show as much info as `ps`. For example, the user who is running a specific process.

Comment (marcelruland) [reply pages/common/ps.md:16]: > There is actually this clever trick to enclose the first letter in `[]` to filter out `grep` itself. Let's see what other maintainers think about this change.

I had considered this as well, but decided against it because it provides info about `grep` on the page about `ps`. It also doesn't explain _why_ `[t]his` works, and adding an explanation would put even more `grep` info on the `ps` page.

Comment (vitorhcl) [reply pages/common/ps.md:16]: > I had considered this as well, but decided against it because it provides info about `grep` on the page about `ps`. It also doesn't explain _why_ `[t]his` works, and adding an explanation would put even more `grep` info on the `ps` page.

IMO it's fine to use other tools in any way (even a complex one) and don't explain _why_ that works.

After all, users can always consult the grep page for more information, and IMO we shouldn't restrict ourselves on the integration of different commands as long as each page focus on explain its own command.

I personally support adding the brackets.

File: pages/common/ps.md
Comment (vitorhcl) [inline pages/common/ps.md:14]:
```suggestion
- Search for a process that matches a string (the brackets filter out `grep` itself):
```
If we include the brackets, how about this?

File: pages/common/ps.md
Comment (marcelruland) [inline pages/common/ps.md:14]:
```suggestion
- Search for a process that matches a string (the brackets will prevent `grep` from matching itself):
```

This phrasing makes it more clear that the brackets are `grep` specific and not `ps` specific.

File: pages/common/ps.md
Comment (marcelruland) [reply pages/common/ps.md]: > [...] we shouldn't restrict ourselves on the integration of different commands as long as each page focus on explain its own command.

Fair point. Though, come to think of it, one could argue that the entire `grep` example is redundant. The first example is `ps aux` and the third example is the exact same thing plus piping to `grep`. It doesn't add any new information about `ps`.

Comment (vitorhcl) [reply pages/common/ps.md]: > [...] It doesn't add any new information about `ps`.

Interesting statement. Personallly, I disagree it is redundant, the focus of tldr is _how we can use the program in useful ways_ rather than _how it works_ (man pages already do a good job on explaining each option)

**Comment by fazlearefin** [comment]
LGTM.

**Review by fazlearefin — APPROVED** [review]

**Review by vitorhcl — APPROVED** [review]
LGTM

**Review by vitorhcl — COMMENTED** [review]
GTG after @tricantivu's approval

**Review by tricantivu — APPROVED** [review]

**Pull request merged**
